src/app/visit/visit.cpp

```diff
@@ -2,22 +2,28 @@
 #include "clinic_facade.hpp"
 #include "patient.hpp"
 #include "room.hpp"
+#include <stdexcept>
 
 Visit::Visit(const std::shared_ptr<Doctor>& doctor, const std::vector<Treatment>& treatments)
-    : doctor_{doctor}, patient_{}, treatments_{std::move(treatments)}, room_{}, visit_information_{}
+    : doctor_{doctor}, patient_{}, room_{}, treatments_{std::move(treatments)}, visit_information_{}
```

Comment:
Shouldn't we swap shared_ptr to weak_ptr for doctor here?
Then we could perform a check if doctor is not expired.
`doctor_{getNotExpired(doctor)}`

Reply:
I think its not a problem because doctor_ is a weak_ptr field hence constructor of shared_ptr seems correct

Reply:
IMO we can change for weak_ptr here. The check is an interesting option. But what do we do if doctor is expired?
I think that only valid option is to throw exception from the constructor in case of expiration.
It has some merits in my opinion, as we can ensure establishing class invariant in the constructor, ensuring that visit can exist only as a valid object.
If we go this way, I think the constructor should take patient argument as well and we should do the same for patient. After all on creating a visit we should know the patient already and Visit existence makes less sense without patient than without a doctor even.

But taking weak_ptr and checking makes sense only if we intend to throw in case of expiration IMO.
Otherwise, as we take reference to shared_ptr and we store it immediately (converting on the fly) into weak_ptr field, than it makes no much sense.

Reply:
Please disregard my last comment.
On second thought, checking on construction makes no sense IMO:
1) weak_ptr can be valid on construction and few miliseconds later not
2) anyone retrieving weak_ptr will have to check any way if its valid